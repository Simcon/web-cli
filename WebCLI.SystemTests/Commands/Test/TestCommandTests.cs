using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;
using WebCLI.Code.Extensions;

namespace WebCLI.SystemTests.Commands.Test
{
    [UseApprovalSubdirectory("approvals")]
    [UseReporter(typeof(DiffReporter))]
    public class TestCommandTests
    {
        [Test]
        public void TestWithoutOption()
        {
            var args = "test";

            var result = ProcessHelper.StartConsoleApplication(args);

            Approvals.Verify(result.ToJson(Newtonsoft.Json.Formatting.Indented));
        }

        [Test]
        public void TestWithOption()
        {
            var args = "test -io 23";

            var result = ProcessHelper.StartConsoleApplication(args);

            Approvals.Verify(result.ToJson(Newtonsoft.Json.Formatting.Indented));
        }
    }
}
