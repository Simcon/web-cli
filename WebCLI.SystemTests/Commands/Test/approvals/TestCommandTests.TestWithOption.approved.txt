{
  "ExitCode": 0,
  "Error": "",
  "Output": "The value for --int-option is: 23\n"
}